chainer/__init__.py

```diff
@@ -150,6 +150,7 @@ def is_arrays_compatible(arrays):
 global_config.use_ideep = os.environ.get('CHAINER_USE_IDEEP', 'never')
 global_config.lazy_grad_sum = bool(int(
     os.environ.get('CHAINER_LAZY_GRAD_SUM', '0')))
+global_config.dtype = numpy.dtype(os.environ.get('CHAINER_DTYPE', 'float32'))
```

Comment:
It's better to show a nice error message if user specified wrong value.

Reply:
Added a check for the value of CHAINER_DTYPE.